code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -263,14 +263,13 @@
 			if(!recording_recipe)
 				var/reagent = GLOB.name2reagent[reagent_name]
 				if(beaker && dispensable_reagents.Find(reagent))
-					var/datum/reagents/R = beaker.reagents
-					var/free = R.maximum_volume - R.total_volume
-					var/actual = min(amount, (cell.charge * powerefficiency)*10, free)
 
-					if(!cell.use(actual / powerefficiency))
+					var/datum/reagents/holder = beaker.reagents
+					var/to_dispense = max(0, min(amount, holder.maximum_volume - holder.total_volume))
```

Comment:
why not use clamp?

Reply:
Because clamp decides which bounds are which when it's parsed so if I try to use `clamp(amount, 0, holder.maximum_volume - holder.total_volume)` and it somehow ends up with more total volume than maximum volume that value will be use as the min and the 0 will be used as the max.

Reply:
what the fuck why